aws/resource_aws_glue_dev_endpoint.go

```diff
@@ -0,0 +1,529 @@
+		return fmt.Errorf("error setting public_address for Glue Dev Endpoint (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("public_key", endpoint.PublicKey); err != nil {
+		return fmt.Errorf("error setting public_key for Glue Dev Endpoint (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("public_keys", flattenStringSet(endpoint.PublicKeys)); err != nil {
+		return fmt.Errorf("error setting public_keys for Glue Dev Endpoint (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("role_arn", endpoint.RoleArn); err != nil {
+		return fmt.Errorf("error setting role_arn for Glue Dev Endpoint (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("security_configuration", endpoint.SecurityConfiguration); err != nil {
+		return fmt.Errorf("error setting security_configuration for Glue Dev Endpoint (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("security_group_ids", flattenStringSet(endpoint.SecurityGroupIds)); err != nil {
+		return fmt.Errorf("error setting security_group_ids for Glue Dev Endpoint (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("status", endpoint.Status); err != nil {
+		return fmt.Errorf("error setting status for Glue Dev Endpoint (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("subnet_id", endpoint.SubnetId); err != nil {
+		return fmt.Errorf("error setting subnet_id for Glue Dev Endpoint (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("vpc_id", endpoint.VpcId); err != nil {
+		return fmt.Errorf("error setting vpc_id for Glue Dev Endpoint (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("worker_type", endpoint.WorkerType); err != nil {
+		return fmt.Errorf("error setting worker_type for Glue Dev Endpoint (%s): %s", d.Id(), err)
+	}
+
+	tags, err := keyvaluetags.GlueListTags(conn, endpointARN)
+
+	if err != nil {
+		return fmt.Errorf("error listing tags for Glue Dev Endpoint (%s): %s", endpointARN, err)
+	}
+
+	if err := d.Set("tags", tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %s", err)
+	}
+
+	if err := d.Set("yarn_endpoint_address", endpoint.YarnEndpointAddress); err != nil {
+		return fmt.Errorf("error setting yarn_endpoint_address for Glue Dev Endpoint (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("zeppelin_remote_spark_interpreter_port", endpoint.ZeppelinRemoteSparkInterpreterPort); err != nil {
+		return fmt.Errorf("error setting zeppelin_remote_spark_interpreter_port for Glue Dev Endpoint (%s): %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func resourceAwsDevEndpointUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).glueconn
+
+	input := &glue.UpdateDevEndpointInput{
+		EndpointName: aws.String(d.Get("name").(string)),
+	}
+
+	hasChanged := false
+
+	customLibs := &glue.DevEndpointCustomLibraries{}
+
+	if d.HasChange("arguments") {
+		oldRaw, newRaw := d.GetChange("arguments")
+		old := oldRaw.(map[string]interface{})
+		new := newRaw.(map[string]interface{})
+		create, remove := diffStringMaps(old, new)
```

Comment:
create, remove, _ := diffStringMaps(old, new)

Reply:
see CI fail

Reply:
I see, the 3 return values were a recent changes, I haven't rebased yet.